safe_transaction_service/history/indexers/safe_events_indexer.py

```diff
@@ -485,3 +478,179 @@ def _process_decoded_element(
+                    setup_args["_owners"] = setup_args.pop("owners")
+                    internal_tx_decoded = InternalTxDecoded(
+                        internal_tx=internal_tx,
+                        function_name="setup",
+                        arguments=setup_args,
+                    )
+                    internal_txs.append(internal_tx)
+                    internal_decoded_txs.append(internal_tx_decoded)
+                elif event["event"] == "ProxyCreation":
+                    proxy_creation_event = event
+                    # Generate InternalTx for ProxyCreation
+                    internal_tx = self._get_internal_tx_from_decoded_event(
+                        proxy_creation_event,
+                        contract_address=proxy_creation_event["args"].get("proxy"),
+                        tx_type=InternalTxType.CREATE.value,
+                        call_type=None,
+                    )
+                    internal_txs.append(internal_tx)
+                else:
+                    logger.error(f"Event is not a Safe creation event {event['event']}")
+
+        # Store which internal txs where inserted into database
+        stored_internal_txs: List[InternalTx] = []
+        with transaction.atomic():
+            try:
+                InternalTx.objects.bulk_create(internal_txs)
+                InternalTxDecoded.objects.bulk_create(internal_decoded_txs)
+                stored_internal_txs = internal_txs
+            except IntegrityError:
+                pass
+
+        if internal_txs and not stored_internal_txs:
+            # Fallback handler in case of integrity error
+            with transaction.atomic():
+                # Skip previous inserted instances
+                for internal_tx in internal_txs:
+                    try:
+                        InternalTx.save(internal_tx)
+                        stored_internal_txs.append(internal_tx)
+                    except IntegrityError:
+                        logger.info(
+                            "Ignoring already processed InternalTx with tx-hash=%s and trace-address=%s",
+                            internal_tx.ethereum_tx_id.hex(),
+                            internal_tx.trace_address,
+                        )
+                        pass
+                for internal_decoded_tx in internal_decoded_txs:
+                    try:
+                        InternalTxDecoded.save(internal_decoded_tx)
+                    except IntegrityError:
+                        logger.info(
+                            "Ignoring already processed InternalDecodedTx with tx-hash=%s and trace-address=%s",
+                            internal_decoded_tx.internal_tx.ethereum_tx_id.hex(),
+                            internal_decoded_tx.internal_tx.trace_address,
+                        )
+                        pass
+
+        return stored_internal_txs
+
+    def _process_decoded_elements(self, decoded_elements: list[EventData]) -> List[Any]:
```

Comment:
```suggestion
    def _process_decoded_elements(self, decoded_elements: list[EventData]) -> list[Any]:
```

If you are using `list` as type, you should use it everywhere 😉